libs/UGridNET/dll/CMakeLists.txt

```diff
@@ -0,0 +1,67 @@
+project(
+  UGridNET
+  VERSION ${CMAKE_PROJECT_VERSION}
+  DESCRIPTION "UGrid C# API"
+  LANGUAGES CSharp
+)
+
+# target name
+set(TARGET_NAME ${PROJECT_NAME})
+
+set(APP_CONFIG ${CMAKE_CURRENT_BINARY_DIR}/App.config)
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/../config/App.config.in
+  ${APP_CONFIG}
+  @ONLY
+)
+
+set(NUGET_CONFIG ${CMAKE_CURRENT_BINARY_DIR}/NuGet.config)
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/../config/NuGet.config
+  ${NUGET_CONFIG}
+  COPYONLY
+)
+
+set(DIRECTORY_BUILD_PROPS ${CMAKE_CURRENT_BINARY_DIR}/Directory.Build.props)
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/../config/Directory.Build.props.in
+  ${DIRECTORY_BUILD_PROPS}
+  @ONLY
+)
+
+set(DIRECTORY_PACKAGES_PROPS ${CMAKE_CURRENT_BINARY_DIR}/Directory.Packages.props)
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/Directory.Packages.props
+  ${DIRECTORY_PACKAGES_PROPS}
+  COPYONLY
+)
+
+execute_process(
+  COMMAND ${Python_EXECUTABLE}
+    ${CMAKE_SOURCE_DIR}/scripts/get_package_ids_from_directory_packages_props.py
+    ${DIRECTORY_PACKAGES_PROPS}
+  OUTPUT_VARIABLE PACKAGE_REFERENCES
+  RESULT_VARIABLE RETURN_CODE
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  COMMAND_ERROR_IS_FATAL LAST
+)
+
+add_library(
+  ${TARGET_NAME}
+  SHARED 
+  ${APP_CONFIG}
+  ${NUGET_CONFIG}
+  ${DIRECTORY_PACKAGES_PROPS}
+  ${DIRECTORY_BUILD_PROPS}
+  ${SWIG_GENERATED_CSHARP_SRCS}
```

Comment:
This variable is defined in the cmake project that runs before (add_subdirectory(SWIG)) and not used there. is it possible to define SWIG_GENERATED_CSHARP_SRCS  here?


file(GLOB SWIG_GENERATED_CSHARP_SRCS CONFIGURE_DEPENDS "${SWIG_OUTPUT_DIR}/*.cs")
set(SWIG_GENERATED_CSHARP_SRCS ${SWIG_GENERATED_CSHARP_SRCS} PARENT_SCOPE)

on the other hand, i understand that the swig project needs to run first

Reply:
You can do file glob in one of either of the 2 configs. It does not matter. The path must be adapted. It makes sense though to create a list of the source in the config where they are generated.